### PR TITLE
feat: add support for polygon-zkevm networks

### PR DESCRIPTION
**Pull request by antazoey**
self explanatory

also required fixing something with multiname network names in API key check

also was forced to have weird handling for not-implemented on a couple methods because was just getting empty 400s and couldnt really do anything with those

### CODE REVIEW

File: ape_alchemy/provider.py [hunk 1]
```diff
@@ -85,6 +98,14 @@ def ws_uri(self) -> str:
         # NOTE: Overriding `Web3Provider.ws_uri` implementation
         return "ws" + self.uri[4:]  # Remove `http` in default URI w/ `ws`
 
+    @property
+    def priority_fee(self) -> int:
+        if self.network.ecosystem.name == "polygon-zkevm":
```
Comment (antazoey) [inline ape_alchemy/provider.py:103]: It appears that Alchemy is very limited with its RPCs for polygon zkevm for some reason.
Using a random node from evmchains works much better.

Comment (antazoey) [reply ape_alchemy/provider.py:103]: ya they just support too many ETH RPCs

**Comment by antazoey** [comment]
a TODO might be to add more support for `zkEVM` Methods

```
[zkevm_consolidatedBlockNumber](https://docs.alchemy.com/reference/zkevm-consolidatedblocknumber-polygon-zkevm)
[zkevm_isBlockConsolidated](https://docs.alchemy.com/reference/zkevm-isblockconsolidated-polygon-zkevm)
[zkevm_isBlockVirtualized](https://docs.alchemy.com/reference/zkevm-isblockvirtualized-polygon-zkevm)
[zkevm_batchNumberByBlockNumber](https://docs.alchemy.com/reference/zkevm-batchnumberbyblocknumber-polygon-zkevm)
[zkevm_batchNumber](https://docs.alchemy.com/reference/zkevm-batchnumber-polygon-zkevm)
[zkevm_virtualBatchNumber](https://docs.alchemy.com/reference/zkevm-virtualbatchnumber-polygon-zkevm)
[zkevm_verifiedBatchNumber](https://docs.alchemy.com/reference/zkevm-verifiedbatchnumber-polygon-zkevm)
[zkevm_getBatchByNumber](https://docs.alchemy.com/reference/zkevm-getbatchbynumber-polygon-zkevm)
[zkevm_getBroadcastURI](https://docs.alchemy.com/reference/zkevm-getbroadcasturi-polygon-zkevm)
```

**Review by fubuloubu — APPROVED** [review]

**Comment by fubuloubu** [comment]
> a TODO might be to add more support for `zkEVM` Methods
> 
> [zkevm_consolidatedBlockNumber](https://docs.alchemy.com/reference/zkevm-consolidatedblocknumber-polygon-zkevm) [zkevm_isBlockConsolidated](https://docs.alchemy.com/reference/zkevm-isblockconsolidated-polygon-zkevm) [zkevm_isBlockVirtualized](https://docs.alchemy.com/reference/zkevm-isblockvirtualized-polygon-zkevm) [zkevm_batchNumberByBlockNumber](https://docs.alchemy.com/reference/zkevm-batchnumberbyblocknumber-polygon-zkevm) [zkevm_batchNumber](https://docs.alchemy.com/reference/zkevm-batchnumber-polygon-zkevm) [zkevm_virtualBatchNumber](https://docs.alchemy.com/reference/zkevm-virtualbatchnumber-polygon-zkevm) [zkevm_verifiedBatchNumber](https://docs.alchemy.com/reference/zkevm-verifiedbatchnumber-polygon-zkevm) [zkevm_getBatchByNumber](https://docs.alchemy.com/reference/zkevm-getbatchbynumber-polygon-zkevm) [zkevm_getBroadcastURI](https://docs.alchemy.com/reference/zkevm-getbroadcasturi-polygon-zkevm)

Not too terribly interested in diving down that rabbit hole unless people really need it

**Pull request merged**